src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -168,13 +168,17 @@ private void OnEventSourceChanged(object? _1, TaggerEventArgs _2)
             => EnqueueWork(highPriority: false);
 
         private void EnqueueWork(bool highPriority)
-            => _eventChangeQueue.AddWork(highPriority, _dataSource.CancelOnNewWork);
+            => EnqueueWork(highPriority, _dataSource.SupportsFrozenPartialSemantics);
 
-        private ValueTask<VoidResult> ProcessEventChangeAsync(ImmutableSegmentedList<bool> changes, CancellationToken cancellationToken)
+        private void EnqueueWork(bool highPriority, bool frozenPartialSemantics)
+            => _eventChangeQueue.AddWork((highPriority, frozenPartialSemantics), _dataSource.CancelOnNewWork);
+
+        private async ValueTask ProcessEventChangeAsync(ImmutableSegmentedList<(bool highPriority, bool frozenPartialSemantics)> changes, CancellationToken cancellationToken)
         {
             // If any of the requests was high priority, then compute at that speed.
-            var highPriority = changes.Contains(true);
-            return new ValueTask<VoidResult>(RecomputeTagsAsync(highPriority, cancellationToken));
+            var highPriority = changes.Contains(x => x.highPriority);
+            var frozenPartialSemantics = changes.Contains(t => t.frozenPartialSemantics);
```

Comment:
>); [](http://example.com/codeflow?start=87&length=2)

if we have one request of (highPriority: true, frozenPartialSemantics: false) and another (highPriority: false, frozenPartialSemantics: true), then we'd recompute with a mix setting of (true, true), is that intended? If so, pls add comment to clearify that. Thanks!

Reply:
will add comment.